packages/splitio-adapter/modules/adapter/adapter.ts

```diff
@@ -220,9 +228,23 @@ class SplitioAdapter implements SplitioAdapterInterface {
       );
     }
 
+    let hasDifference = false;
```

Comment:
Nit: It feels easier to not have a `let` here but instead 

```js
const hasUserChanged = !isEqual(adapterState.user, adapterArgs.user);
const hasTreatmentChanged = !isEqual(
  adapterState.treatmentAttributes,
  adapterArgs.treatmentAttributes
);

if (hasUserChanged) { adapterState.user = ensureUser(adapterArgs.user); }
if (hasTreatmentChanged) { adapterState.treatmentAttributes = adapterArgs.treatmentAttributes; }

if (hasUserChanged && hasTreatmentChanged && adapterState.manager && adapterState.client) {
   adapterState.client.destroy();
}

Reply:
hmm, that probably needs to be `(hasUserChanged || hasTreatmentChanged && (adapterState.manager && adapterState.client))` as we need to destroy the client if either of those change. But I agree, this looks nicer.

Reply:
True. Sorry.

Reply:
No need to apologize. Appreciate the feedback :)

Updated.